File: src/1432-Max-Difference-You-Can-Get-From-Changing-an-Integer/1432.py
class Solution:
    def maxDiff(self, num: int) -> int:
        a = b = str(num)

        for digit in a:
            if digit != "9":
                a = a.replace(digit, "9")
                break
        
        if b[0] != "1":
            b = b.replace(b[0], "1")
        else:
            for digit in b[1:]:
                if digit not in "01":
                    b = b.replace(digit, "0")
                    break
        
        return int(a) - int(b)

File: src/1432-Max-Difference-You-Can-Get-From-Changing-an-Integer/1432.cpp
class Solution {
public:
    int maxDiff(int num) {
        string a = to_string(num), b = a;
        for (char digit : a) {
            if (digit != '9') {
                replace(a.begin(), a.end(), digit, '9');
                break;
            }
        }
        
        if (b[0] != '1') {
            char c = b[0];
            replace(b.begin(), b.end(), c, '1');
        } else {
            for (int i = 1; i < b.size(); i++) {
                if (b[i] != '0' && b[i] != '1') {
                    char c = b[i];
                    replace(b.begin(), b.end(), c, '0');
                    break;
                }
            }
        }
        return stoi(a) - stoi(b);
    }
};